File: BankingApplicationSolution/BankingLibrary/Account.cs
using System;

namespace BankingLibrary {

    public abstract class Account {

        private static int NextAcctNbr = 1;
        private const int AcctNbrInc = 9;

        public int AcctNbr { get; private set; }
        public string Description { get; set; } = "Account";
        public decimal Balance { get; set; }

        private int AttemptsToOverdraw = 0;

        private bool CheckAmountGTZero(decimal amount) {
            return (amount <= 0) ? false : true;
        }

        public void Deposit(decimal amount) {
            if(CheckAmountGTZero(amount)) {
                Balance += amount;
            }
        }

        private bool IsSufficientFunds(decimal amount) {
            if(Balance >= amount) {
                return true;
            }
            AttemptsToOverdraw++;
            return false;
        }
        public bool Withdraw(decimal amount) {
            if(CheckAmountGTZero(amount) && IsSufficientFunds(amount)) {
                Balance -= amount;
                return true;
            }
            return false;
        }
        public void Transfer(decimal amount, Account toAccount) {
            if(this.Withdraw(amount)) {
                toAccount.Deposit(amount);
            }
        }

        public override string ToString() {
            return $"AcctNbr={AcctNbr}, Desc={Description}, Bal={Balance}";
        }

        public void Debug() {
            Console.WriteLine(this);
        }

        public Account() {
            this.AcctNbr = NextAcctNbr;
            NextAcctNbr += AcctNbrInc;
        }
    }
}

File: BankingInJava/src/com/maxtrain/bootcamp/dotnet/dnbc8/Account.java
package com.maxtrain.bootcamp.dotnet.dnbc8;

public abstract class Account implements IAccount {
	
	private static int nextId = 1;
	private int id;
	private String description;
	private double balance;
	
	public void print() {
		System.out.println(this.toString());
	}
	private boolean isAmountGtZero(double amount) {
		return amount > 0;
	}
	private boolean isSufficientFunds(double amount) {
		return balance >= amount;
	}
	public boolean deposit(double amount) {
		if(!isAmountGtZero(amount))
			return false;
		balance += amount;
		return true;
	}
	public boolean withdraw(double amount) {
		if(!isSufficientFunds(amount) || !isAmountGtZero(amount))
			return false;
		balance -= amount;
		return true;
	}
	public boolean transfer(double amount, Account toAccount) {
		boolean success = withdraw(amount);
		if(!success) {
			return false;
		}
		toAccount.deposit(amount);
		return true;
	}
	@Override
	public String toString() {
		return String.format("%02d/%-17s/%-9.2f", getId(), getDescription(), getBalance());
	}
	public Account(String description) {
		setId(nextId++);
		setDescription(description + " " + nextId);
		setBalance(0);
	}
	public Account() {
		this("New Account");
	}
	/*
	// getters & setters
	*/
	public int getId() {
		return id;
	}
	protected void setId(int id) {
		this.id = id;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public double getBalance() {
		return balance;
	}
	protected void setBalance(double balance) {
		this.balance = balance;
	} 
}
